### PR TITLE
Implement Bga component 

### PR DESCRIPTION
**Pull request by abhijitxy**
#15

### CODE REVIEW

**Comment by abhijitxy** [comment]
@seveibar please review it:)

**Comment by seveibar** [comment]
Looks pretty good, but can you add an example red footprint eg "bga64" just so we can check orientation

Also can you make the chip's body the same color as ChipBody, like a dark gray


image

**Comment by seveibar** [comment]
Btw the code is extremely good imo, a perfect implementation of the diagram.

We may want to rename the vars though so people dont need to look up the diagram

**Comment by abhijitxy** [comment]
> Looks pretty good, but can you add an example red footprint eg "bga64" just so we can check orientation
> 
> Also can you make the chip's body the same color as ChipBody, like a dark gray
> 
> image

sounds good!

File: lib/BGA.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if (!footprintString) return null;
+    const result = fp.string(footprintString);
+    return result.soup();
+  }, [footprintString]);
+
+  return (
+    <>
+      {/* Package body */}
+      <Colorize color="#555">
+        <Rotate rotation={[190, 0, 0]}>
```
Comment (seveibar) [inline lib/BGA.tsx:44]:
```suggestion
        <Rotate rotation={[180, 0, 0]}>
```

Comment (seveibar) [reply lib/BGA.tsx:44]: We discussed on discord that this is maybe an issue with `jscad-fiber` where we need to make things multiples of `Math.PI`/`Math.PI/2`

File: examples/BGA.example.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { JsCadFixture } from "jscad-fiber";
+import { BGA } from "../lib/BGA";
+import { ExtrudedPads } from "../src/lib/ExtrudedPads";
+
+export default () => {
+  return (
+    <JsCadFixture zAxisUp>
+        <BGA footprintString="bga64" />
```
Comment (seveibar) [inline examples/BGA.example.tsx:8]:
```suggestion
      <BGA footprintString="bga64" />
```

**Review by seveibar — APPROVED** [review]

**Pull request merged**